### PR TITLE
Add a new APIv1 endpoint to delete dataset fields

### LINKED ISSUE #2771: Allow deletion of dataset fields via API
We should allow deletion of dataset fields before the dataset has been published. With this users can easily delete dataset fields with errors.

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR add changes to allow the deletion of fields specifying the id of a field resource.

Notes:
* I have modified annotations deletion endpoint to return an `Annotation` schema including the `dataset_id` as we are doing with `Field`. 

Closes #2771 

**WARNING:** You should review and merge PR #2770 before review this one.

### CODE REVIEW

File: src/argilla/server/schemas/v1/annotations.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+from pydantic import BaseModel
+
+
+class Annotation(BaseModel):
+    id: UUID
+    name: str
+    title: str
+    required: bool
+    settings: dict
```
Comment (frascuchon) [inline src/argilla/server/schemas/v1/annotations.py]: Should not annotation settings be a compound attribute of available question settings that we have already defined?

File: src/argilla/server/schemas/v1/fields.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+from pydantic import BaseModel
+
+
+class Field(BaseModel):
+    id: UUID
+    name: str
+    title: str
+    required: bool
+    settings: dict
```
Comment (frascuchon) [inline src/argilla/server/schemas/v1/fields.py]: `TextFieldSettings` instead? Regarding generated from pydantic schema, makes sense to provide this info to API consumers.

File: src/argilla/server/schemas/v1/annotations.py (hunk 1 above)
Comment (jfcalvo) [reply src/argilla/server/schemas/v1/annotations.py]: Done.

File: src/argilla/server/schemas/v1/fields.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla/server/schemas/v1/fields.py]: Done.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**